cli/xlsx2json.py

```diff
@@ -0,0 +1,250 @@
+        :return: True if all the worksheets contain required headers. False otherwise
+        :rtype: bool
+        """
+        if self.valid is None:
+            self.valid = True
+            self.valid_worksheets()
+
+        return self.valid
+
+    @staticmethod
+    def cast_value(value, type_name):
+        # Do not cast None values
+        if type_name and value is not None:
+            if type_name == 'string':
+                return str(value)
+        return value
+
+    @staticmethod
+    def trim_value(value):
+        """Remove whitespace from the start and end of cells"""
+        if isinstance(value, str):
+            return value.strip()
+        return value
+
+    def __iter__(self):
+        return self
+
+    def base_row_offset(self, worksheet):
+        return self.xls_conf[worksheet].get(HEADERS_KEY_ROW, 1)
+
+    def get_rows(self):
+        """
+        Retrieve all the data rows.
+        :return: list of hash containing all the REQUIRED and OPTIONAL fields as keys
+                and the corresponding data as values
+        :rtype: list
+        """
+        worksheet = self.active_worksheet
+        if worksheet is None:
+            logging.warning('No worksheet is specified!')
+            return None
+
+        if worksheet not in self.row_offset:
+            self.row_offset[worksheet] = self.base_row_offset(worksheet)
+        self.row_offset[worksheet] += 1
+
+        required_headers = self.xls_conf[worksheet].get(REQUIRED_HEADERS_KEY_NAME, {})
+        optional_headers = self.xls_conf[worksheet].get(OPTIONAL_HEADERS_KEY_NAME, {})
+
+        rows = []
+
+        for row in self.workbook[worksheet].iter_rows(min_row=self.row_offset[worksheet]):
+            num_cells = 0
+            for cell in row:
+                num_cells += 1
+
+            data = {}
+            has_notnull = False
+            for header in list(required_headers.keys()) + list(optional_headers.keys()):
+                header_index = num_cells
+                if header in self.headers[worksheet]:
+                    header_index = self.headers[worksheet].index(header)
+                if header_index >= num_cells:
+                    data[header] = None
+                    continue
+
+                cell = row[header_index]
+                if cell.value is not None:
+                    has_notnull = True
+                data[header] = self.trim_value(self.cast_value(
+                    cell.value, self.xls_conf[worksheet].get(CAST_KEY_NAME, {}).get(header)
+                ))
+
+            if has_notnull:
+                data['row_num'] = self.row_offset[worksheet]
+                rows.append(data)
+
+            # no data on this row, continue to next
+            self.row_offset[worksheet] += 1
+
+        return rows
+
+    def json(self, output_json_file):
+        json_data = {}
+        for title in self.xls_conf[WORKSHEETS_KEY_NAME]:
+            json_data[self.xls_conf[WORKSHEETS_KEY_NAME][title]] = []
```

Comment:
Looking at `eva_schema.json` I guess `project` is an object not an array, so that sheet will need to be handled differently here.

Reply:
Did you check this?  May be a misunderstanding on my part, just referring to the type of project [here](https://github.com/EBIvariation/eva-sub-cli/blob/main/cli/etc/eva_schema.json#LL67C4-L67C4) being the only one that's object and not array

Reply:
I did. I added it to EVA-3293's scope. I'm keeping the scope of this ticket to literally the direct translation of the spreadsheet.